### DISCUSSION TITLE
Here I will show my experience to run Adwaita on Embedded Linux

### DISCUSSION THREAD

**Discussion by CmST0us** [AparokshaUI/Adwaita#4] [Show and tell]
Thanks to this great work! Finally I can write GUI App on my embedded linux platform.
So here, I will show my progress of running Adwaita on embedded linux.

The goal is to write a Linux App, which can capture HDMI-IN image, and publish the image to mobile phone or vision pro App. And I will use Swift language in all module, include Video Capture, Image Codec, Publish, Render.
With Adwaita, I can write GUI App, which help me mix the control element like Start/Stop Button, Publish Info, and more

Below is the Adwaita related step.
- [x] Use Yocto to build a sdk that include libadwaita.
- [x] Create Swift toolchain that can cross-compile Adwaita.
- [x] Cross-compile Adwaita and run on target platform.
- [ ] Try to adapt and run on 32-bit embedded linux.

**Comment by CmST0us** [comment]
Finally I create my yocto toolchain that can help us cross compile adwaita-swift!
![2024-04-06 23-52-08屏幕截图](https://github.com/AparokshaUI/adwaita-swift/assets/16432714/adae6caa-a3e7-46e4-a4e9-4671e794d5fd)

**Reply by CmST0us** [reply]
Here is what adwaita-swift run on rk3566 weston looks like.
![IMG_4301](https://github.com/AparokshaUI/adwaita-swift/assets/16432714/acc93c7b-639c-4979-9191-f988bea087f3)

**Reply by CmST0us** [reply]
It looks like Label will layout in a small size..
But the official has the same problem.

**Comment by CmST0us** [comment]
So next,  I will change my platform from 64-bit aarch64 ti 32-bit armv7.
and try to run adwaita.
I think it will take a long time....

**Reply by david-swift** [reply]
Thank you so much for sharing your experiences @CmST0us!
